docs/blog/20230501-Daeraxa-MayUpdate.md

```diff
@@ -0,0 +1,161 @@
+
+<img src="./assets/octicons2.png"/>
+
+We have a couple of different goals here. The main one is simply to give more options to ourselves and users in terms of icon variation so that we can make better choices for icon usage. We would of course retain the `4.4.0` set (just as `2.1.2` has also been retained).
+
+One of our others is a plan to hopefully modernise the UI, for example we have a bunch of new icons we can use for extending the set of file icons. You can see a couple of potential implementations in the below picture compared to the original. We actually have a survey up right now to gauge the community response to this so if you haven't already then please consider giving us [your opinion](https://docs.google.com/forms/d/e/1FAIpQLSfnEEAQHUJbRVcH2NlNvLxmK0ZUcb5xuOCt1pYOGytvne0kJw/viewform?usp=sf_link).
+
+<img src="./assets/octicons3.png"/>
+
+There is still some work to go here, particularly on making sure that where icons are used we are still using the appropriate ones or selecting a more appropriate icon now it is available in the new set.
+
+If you want to read more details about what had to be done for the implementation then you can read the [document](https://github.com/pulsar-edit/pulsar/blob/octicons-v18.3.0/static/icons/README.md) that accompanies the PR that details what was done and how to maintain it in the future.
+
+## Symbols view improvements
+
+<img src="./assets/symbols-view.png"/>
+
+Currently in order to use Pulsar's symbols view for navigation you first have to generate a [ctags](https://ctags.io/) file for your project. This isn't really a perfect solution as it doesn't allow navigation of untitled files and requires the file to be updated every time there is a significant refactor of your code.
+
+[@savetheclocktower] has been working on some improvements for this, for example for languages that have tree-sitter parsers, its job can be done much better - all it does is answer the question “what are the names and line numbers for important things like class and method definitions?”
+
+Currently this is being explored due to the ongoing work to upgrade to a modern implementation of `tree-sitter` but the eventual goal would be to move to a provider/consumer model like our `autocomplete-plus` package which would allow for much more flexibility, for example [language servers](https://microsoft.github.io/language-server-protocol/) are also potentially able to provide symbols, and on a project-wide basis - not just within a file.
+
+So to make steps towards achieving this, step one would be to refactor `symbols-view` to be more generic then step two would be to create packages called, for example, `symbol-provider-ctags` and `symbol-provider-tree-sitter` - both of which would be built-in packages.
```

Comment:
I would avoid mentioning the step 2's details. Mainly because they can change - for example, I wasn't expecting to split symbols-view into different providers, so I'm not sure if it'll actually be done

Reply:
Maybe @savetheclocktower could weigh in on this one as the two statements are a little contradictory.

Reply:
> I wasn't expecting to split symbols-view into different providers, so I'm not sure if it'll actually be done

You mean you wanted to leave `ctags` in `symbols-view` but still have it consume other providers? That's fine, but I did circulate a design document about a week ago and got no objections, so… this code is mostly written.

But, yes, it's still a good idea to be a little more vague in this paragraph.